cli/azd/pkg/tools/dotnet/dotnet.go

```diff
@@ -205,18 +206,31 @@ func (cli *dotNetCli) PublishContainer(
 		return 0, fmt.Errorf("dotnet publish on project '%s' failed: %w", project, err)
 	}
 
-	port, err := cli.getTargetPort(result.Stdout)
+	port, err := cli.getTargetPort(result.Stdout, project)
 	if err != nil {
 		return 0, fmt.Errorf("failed to get dotnet target port: %w", err)
 	}
 
 	return port, nil
 }
 
-func (cli *dotNetCli) getTargetPort(result string) (int, error) {
+func (cli *dotNetCli) getTargetPort(result, project string) (int, error) {
 	var targetPorts []targetPort
 	var configOutput responseContainerConfiguration
 
+	// make sure result only contains json config output
+	result = strings.Split(result, "\n\r\n\r")[0]
+
+	// if empty string or there's no config output
+	if result == "" || !strings.Contains(result, "{\"config\":") {
+		return 0, &azcli.ErrorWithSuggestion{
+			Err: fmt.Errorf("empty dotnet configuration output"),
+			Suggestion: fmt.Sprintf("project '%s' isn't enabled for container support. To enable SDK "+
```

Comment:
`EnableSdkContainerSupport ` Could be one of the reasons for an empty dotnet config? Should we also surface the error from the dotnet command too?